### PR TITLE
1895 new rep locator over master

### PR DESCRIPTION
**Pull request by duncanparkes**
Split MPs and MPLs into two tabs.

Fixes #2017.

### CODE REVIEW

**Comment by mhl** [comment]
As mentioned in IRC, the Javascript that adds the markers, adds the ward boundary and changes zoom when switching tabs needs to be updated.

File: pombola/south_africa/templates/south_africa/local_rep_list_item.html [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+      <p class="constituency-office-mp-name">{{ person_data.name }}</p>
+    {% if person_data.person %}
+      </a>
+    {% endif %}
+
+    {# This person's position (either as an MP, an MPL or a ward councillor #}
+    {% if not skip_positions %}
+      <ul class="unstyled-list">
+      {% for position in person_data.positions %}
+        {% if position.title.name == 'Ward Councillor' %}
```
Comment (duncanparkes) [inline pombola/south_africa/templates/south_africa/local_rep_list_item.html]: Perhaps better to have a lookup of preserved titles?

Pass in a `preserved_titles` set `{'Ward Councillor', 'Constituency Contact'}` and do '{% if if position.title.name in preserved_titles %}<li>position.title.name</li>{% else %}'

File: pombola/south_africa/templates/south_africa/local_rep_list_item.html [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+    {# Display a photo if one is available... #}
+    {% if person_data.person %}
+      <a href="{{ person_data.person.get_absolute_url }}">
+      {% thumbnail person_data.person.primary_image "58x58" crop="center" as im %}
+        <img class="constituency-office-mp-photo" src="{{ im.url }}" width="{{ im.width }}" height="{{ im.height }}"/>
+      {% empty %}
+        <img class="constituency-office-mp-photo" src="{% static 'images/person-90x90.jpg' %}" width="58" height="58" />
+      {% endthumbnail %}
+    {% else %}
+        <img class="constituency-office-mp-photo" src="{% static 'images/person-90x90.jpg' %}" width="58" height="58" />
```
Comment (duncanparkes) [inline pombola/south_africa/templates/south_africa/local_rep_list_item.html]: What does it mean if there is no `person_data.person`? Should we still use a blank person image, or something else?

File: pombola/south_africa/templates/south_africa/local_rep_list_item.html [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+
+    {% if person_data.email %}
+      <p>Email: {{ person_data.email }}</p>
+    {% endif %}
+
+    {% if person_data.phone %}
+      <p>Phone: {{ person_data.phone }}</p>
+    {% endif %}
+
+    {# Get the constituency office details, if we have theM: #}
```
Comment (duncanparkes) [inline pombola/south_africa/templates/south_africa/local_rep_list_item.html]: Strange capital M!

File: pombola/south_africa/views.py [hunk 4]
```diff
@@ -144,87 +144,133 @@ def get_object(self):
...
+        # should only be one at the moment, but make it a list in case
+        # we support broader lookups in the future:
+        url_fmt = 'http://nearby.code4sa.org/councillor/ward-{ward_id}.json'
+        r = requests.get(url_fmt.format(ward_id=ward_id))
+        ward_result = r.json()
+        councillor_data = ward_result['councillor']
+        party = models.Organisation.objects.filter(
+            name__icontains=councillor_data['PartyDetail']['Name'].lower(),
+            kind__slug='party'
+        ).first()
```
Comment (duncanparkes) [inline pombola/south_africa/views.py]: I assume it's better to pick the first if there is more than one than to use a `.get()`?

**Comment by duncanparkes** [comment]
This work has now been subsumed into #1965

**Pull request closed without merging**